spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanDependencyResolverFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.spring.config.annotation;
+
+import org.axonframework.common.Priority;
+import org.axonframework.eventhandling.AnnotationEventHandlerAdapter;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.beans.factory.config.DependencyDescriptor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.MethodParameter;
+import org.springframework.lang.NonNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+
+/**
+ * ParameterResolverFactory implementation that resolves parameters using Spring dependency resolution.
+ * Mark a parameter as {@link org.springframework.beans.factory.annotation.Autowired} to resolve said parameter
+ * using Spring dependency resolution.
+ *
+ * @author Joel Feinstein
+ * @since 4.5
+ * @see Autowired
+ */
+@Priority(Priority.LOW)
+public class SpringBeanDependencyResolverFactory implements ParameterResolverFactory {
+
+    private final ApplicationContext applicationContext;
+
+    public SpringBeanDependencyResolverFactory(@NonNull ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public ParameterResolver<?> createInstance(Executable executable, Parameter[] parameters, int parameterIndex) {
+        final Autowired ann = parameters[parameterIndex].getAnnotation(Autowired.class);
```

Comment:
Although this works, we aim to allow meta-annotations too.
If my mind serves me right, `Parameter#getAnnotation(Class<T>)` will only return if the given annotation is actually present.

Slightly debatable if it makes sense for Spring `@Autowired` annotation, as the behaviour is mainly aimed so that users are not required to uses Axon's `@CommandHandler` annotation directly for example.
Still, I feel there's a benefit to allowing this.

Luckily, Axon Framework has an `AnnotationUtils` class for this, which does check for meta-annotations as well.
By using `AnnotationUtils#isAnnotationPresent(AnnotatedElement, Class<? extends Annotation>)` you will receive a `boolean` if the `Autowired` annotation is present on the parameter, or on an annotation on the parameter.
You can see the `AbstractAnnotatedParameterResolverFactory` to get a feel how the framework uses this for parameter resolvers.